src/main/include/autos/AutoFactory.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <frc/DriverStation.h>
+#include <frc/Filesystem.h>
+#include <frc2/command/CommandPtr.h>
+#include <frc2/command/WaitCommand.h>
+#include <pathplanner/lib/commands/FollowPathHolonomic.h>
+#include <wpi/MemoryBuffer.h>
+
+#include <filesystem>
+#include <functional>
+
+#include "Constants.h"
+#include "utils/ConsoleLogger.h"
+
+using namespace AutoConstants;
+
+namespace AutoFactory {
+frc2::CommandPtr GetEmptyCommand() { return frc2::WaitCommand(15_s).ToPtr(); }
+
+bool AutoFileExists(std::string fileName) {
+  const std::string filePath = frc::filesystem::GetDeployDirectory() +
+                               "/pathplanner/autos/" + fileName + ".auto";
+
+  std::error_code error_code;
+  std::unique_ptr<wpi::MemoryBuffer> fileBuffer =
+      wpi::MemoryBuffer::GetFile(filePath, error_code);
+
+  if (fileBuffer == nullptr || error_code) {
+    return false;
+  }
+
+  return true;
+}
+
+frc2::CommandPtr PathPlannerPathFromName(std::string autoName) {
+  if (!AutoFileExists(autoName)) {
+    ConsoleLogger::getInstance().logError(
+        "Auto Factory", "AUTO '%s' DOES NOT EXIST HELP US EVAN",
+        autoName.c_str());
+    return GetEmptyCommand();
+  }
+  return pathplanner::PathPlannerAuto(autoName).ToPtr();
+}
+
+frc2::CommandPtr GetAuto(AutoType type) {
+  switch (type) {
```

Comment:
Would this be more extensible as a map?

Reply:
No because the map is initialized only once at startup

Reply:
Fair enough